packages/@aws-cdk/cloudformation-include/build.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * This build file has two purposes:
+ *   1. It adds a dependency on each @aws-cdk/aws-xyz package with L1s to this package,
+ *     similarly to how deps.js does for decdk.
+ *   2. It generates the file cfn-types-2-classes.json that contains a mapping
+ *     between the CloudFormation type and the fully-qualified name of the L1 class,
+ *     used in the logic of the CfnInclude class.
+ */
+
+const fs = require('fs');
```

Comment:
This build logic here, especially generation of the JSON file, is key to the success of this package. I'm worried we don't have unit tests for this logic. And there is cyclomatic complexity in here that needs to be tested.

Wondering we should re-order the build logic so that the generation of cfn-types-2-classes.json happens post unit tests, and then add tests for this.

Reply:
I'm not sure I follow. We have unit tests inside of this package. If there is any error with the generation of the `cfn-types-2-classes.json`, those unit tests will fail.

For that same reason, moving the generation of this file post-unit tests is not an option.

Reply:
The complexity of the service has no bearing on the probability of this file being generated incorrectly. If you look, S3 and KMS are not special in any way (they are not called out by name anywhere), so it doesn't really matter that only those services are tested in the unit tests of this package (so is CodeCommit, by the way).

The only risk here is that a package that should have its L1s in the file will not. I don't think that's very likely, and, even if it does happen, it is not a catastrophic bug.

I'm hard-pressed to even come up with a sensible unit test for this file (what exactly would it check...?).